src/platform/w3dengine/client/w3ddisplay.cpp

```diff
@@ -60,80 +94,457 @@ W3DDisplay::W3DDisplay() :
+        }
+
+#ifdef BUILD_WITH_D3D8
+        if (W3D::Init(g_applicationHWnd)) {
+            throw CODE_07;
+        }
+#endif
+
+        W3D::Set_Prelit_Mode(W3D::PRELIT_MODE_LIGHTMAP_MULTI_PASS);
+        W3D::Set_Collision_Box_Display_Mask(0);
+        W3D::Enable_Static_Sort_Lists(true);
+        W3D::Set_Thumbnail_Enabled(false);
+        W3D::Set_Screen_UV_Bias(true);
+        W3D::Set_Texture_Bit_Depth(32);
+
+        Set_Windowed(g_theWriteableGlobalData->m_windowed);
+
+        m_2DRender = new Render2DClass();
+        captainslog_dbgassert(m_2DRender, "Cannot create Render2DClass");
+
+        Set_Width(g_theWriteableGlobalData->m_xResolution);
+        Set_Height(g_theWriteableGlobalData->m_yResolution);
+        Set_Bit_Depth(32);
+
+        if (W3D::Set_Render_Device(0, Get_Width(), Get_Height(), Get_Bit_Depth(), Get_Windowed(), true, false, true)) {
+            Set_Bit_Depth(16);
+
+            if (W3D::Set_Render_Device(0, Get_Width(), Get_Height(), Get_Bit_Depth(), Get_Windowed(), true, false, true)) {
+                throw CODE_07;
+                captainslog_debug("Unable to set render device");
+            }
+        }
+
+        if (g_theGameLODManager->Get_Static_LOD_Level() == STATLOD_INVALID) {
+            g_theGameLODManager->Set_Static_LOD_Level(g_theGameLODManager->Find_Static_LOD_Level());
+        } else {
+            if (g_theWriteableGlobalData->m_textureReductionFactor) {
+                W3D::Set_Texture_Reduction(g_theWriteableGlobalData->m_textureReductionFactor, 32);
+                g_theGameLODManager->Set_Texture_Reduction_Factor(g_theWriteableGlobalData->m_textureReductionFactor);
+            }
+        }
+
+        if (g_theWriteableGlobalData->m_gammaValue) {
+            Set_Gamma(g_theWriteableGlobalData->m_gammaValue, 0.0f, 1.0f, false);
+        }
+
+        Init_Assets();
+        Init_2D_Scene();
+        Init_3D_Scene();
+        W3DShaderManager::Init();
+
+        m_nativeDebugDisplay = new W3DDebugDisplay();
+        m_debugDisplay = m_nativeDebugDisplay;
+
+        if (m_nativeDebugDisplay) {
+            m_nativeDebugDisplay->Init();
+            GameFont *font;
+
+            if (g_theGlobalLanguage && g_theGlobalLanguage->Debug_Display_Font().Name().Is_Not_Empty()) {
+                font = g_theFontLibrary->Get_Font(g_theGlobalLanguage->Debug_Display_Font().Name(),
+                    g_theGlobalLanguage->Debug_Display_Font().Point_Size(),
+                    g_theGlobalLanguage->Debug_Display_Font().Bold());
+            } else {
+                font = g_theFontLibrary->Get_Font("FixedSys", 8, false);
+            }
+
+            m_nativeDebugDisplay->Set_Font(font);
+            m_nativeDebugDisplay->Set_Font_Width(13);
+            m_nativeDebugDisplay->Set_Font_Height(9);
+        }
+
+        // DX8WebBrowser::Initialize
+        m_initialized = true;
+
+        if (g_theWriteableGlobalData->m_displayDebug) {
```

Comment:
nullptr check

Reply:
fixed.

Reply:
actually this is a bool so it doesn't need a nullptr check